### PR TITLE
[CFM-1255] -Add star toggle to highlighted items on the group dashboard

### PR DESCRIPTION
**Pull request by ybaras**
[CFM-1255](https://issuetracker.amplexor.com/jira/browse/CFM-1255)

### CODE REVIEW

**Comment by ybaras** [comment]
@ordavidil `c4m_features_og_highlights module` is merged. Is this issue still relevant?

**Comment by ordavidil** [comment]
Seems like this one is related to #1202

**Comment by soniCaH** [comment]
Just to make things clear, as I see there are some mix-ups:

There are 2 concepts which will work next to each other:
- Highlighting content within a group (star / unstar icon)
- Promoting content within a group (pin / unpin)

Highlighted content will be listed in a block on the group landing page, and is based on the "promote to frontpage" field of a node. See https://issuetracker.amplexor.com/jira/browse/CFM-1253

Per group 1 node can also be promoted (aka pinned). This node will be displayed on top of the activity stream and is using a flag to toggle the state.

In this PR, we work with the highlighted content (thus the "promote to front page" field), and this icon should be added to the "block_list" view mode:

screen shot 2016-09-20 at 14 55 34

Please also implement this as a JavaScript toggle, instead of a hard link.

**Comment by DavidHernandez** [comment]
### Screenshots:
### Desktop:
highlights-desktop
### Tablet:
highlights-tablet
### Mobile
highlights-mobile

## This issue now depends on feature/CFM-1254-highlight-toggle-fixes

**Comment by ordavidil** [comment]
@DavidHernandez 
please check the tests again:
https://travis-ci.org/capacity4dev/capacity4more/jobs/161860278#L7900

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 1]
```diff
@@ -628,10 +628,15 @@ function c4m_content_node_view($node, $view_mode, $langcode) {
     return;
   }
 
+  $group = og_context();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: use `c4m_og_current_group()` and make sure to validate the returned value.

**Comment by ordavidil** [comment]
@ybaras 
please add screenshots

**Comment by ybaras** [comment]
Icon placement on **widescreen**:

desktop

**Tablet**:

tab

**Smartphone**:

sp

**Comment by ybaras** [comment]
**Reconfigured display with icon**

highlights

**Comment by ybaras** [comment]
Closed in favor of #1439

**Pull request closed without merging**